doc/nrf/ug_bootloader_adding.rst

```diff
@@ -0,0 +1,436 @@
+   Otherwise, you can do the following:
+
+   * On Unix systems, you can symlink the file as :file:`bootloader/mcuboot/priv.pem`, but note that git will see this file as an untracked file within the MCUboot repository.
+
+   * You can create an external project configuration for MCUboot, such as :file:`mcuboot_prj.conf` file in the same directory as the key file and then point to it during build time.
+     If you give the key file as ``CONFIG_BOOT_SIGNATURE_FILE="priv.pem"`` in the external project configuration, it will be used when compiling with ``-Dmcuboot_CONF_FILE=/absolute/path/to/mcuboot_prj.conf``
+
+It is recommended to also set the associated configuration for a key type to ensure MCUboot compiles the public key into its image correctly.
+
+.. code-block:: console
+
+   west build -b nrf52dk_nrf52840 zephyr/samples/hello_world -- \
+   -DCONFIG_BOOTLOADER_MCUBOOT=y \
+   -Dmcuboot_CONFIG_BOOT_SIGNATURE_KEY_FILE=\"../../priv-ecdsa256.pem\" \
+   -Dmcuboot_CONFIG_BOOT_SIGNATURE_TYPE_ECDSA_P256=y
+
+.. note::
+
+   Sometimes the key-type option is automatically applied based upon available hardware.
+   For example, the default nRF52840 development kit configuration will compile signature keys using ECDSA with its *CryptoCell 310* (CC310) hardware.
+   In this case, the following command is valid:
+
+   .. code-block:: console
+
+      west build -b nrf52840dk_nrf52840 zephyr/samples/hello_world -- \
+      -DCONFIG_BOOTLOADER_MCUBOOT=y \
+      -Dmcuboot_CONFIG_BOOT_SIGNATURE_KEY_FILE=\"../../priv-ecdsa256.pem\"
+
+   However, the nRF52832 does not have the CC310 hardware and will instead default to an RSA key.
+   This will cause build errors when using an ECDSA key without enabling its associated key-type option.
+
+You can find specific configuration options for using keys with this bootloader in :file:`bootloader/mcuboot/boot/zephyr/Kconfig`.
+
+Checking the public key
+^^^^^^^^^^^^^^^^^^^^^^^
+
+You can extract the public key locally and compare it against MCUboot's auto-generated file to verify that it is using the custom key:
+
+.. code-block:: console
+
+   diff build/mcuboot/zephyr/autogen-pubkey.c <(python3 bootloader/mcuboot/scripts/imgtool.py getpub -k priv.pem)
+
+If there is no file diff output, then the private key was successfully included with the bootloader image.
+
+.. _ug_bootloader_adding_upgradable:
+
+Adding an upgradable bootloader
+*******************************
+
+MCUboot is the only upgradable bootloader currently available with the |NCS|.
+The following section describes how to add it to your secure bootloader chain.
+
+.. _ug_bootloader_adding_upgradable_mcuboot:
+
+MCUboot
+=======
+
+To use MCUboot as an upgradable bootloader, the application must include the |NSIB| as the immutable bootloader.
+To do so, add :option:`CONFIG_BOOTLOADER_MCUBOOT` with either the build command or in the application's :file:`prj.conf` file:
```

Comment:
Maybe we should mention both options here: `CONFIG_SECURE_BOOT` and `CONFIG_BOOTLOADER_MCUBOOT`? CONFIG_BOOTLOADER_MCUBOOT actually enables only MCUboot

Reply:
@kyberdin

Reply:
@MarekPieta Thanks for suggesting this, my wording here is bad. I have modified it to be clearer about MCUboot's config being used _in addition_ to the immutable's config, as demonstrated in the example code. Since the NSIB's config option is already shown shown and explained earlier in the docs, I didn't include it again in the text here. Please see the latest commit.